deployments/cluster/witness.nix

```diff
@@ -13,44 +13,44 @@ in
   # `map` to make additional SGs easier to add and SG list more readable
   deployment.ec2.securityGroupIds = map (x: resources.ec2SecurityGroups."cluster-${x}-sg".name ) (
     [ "witness-api-private" ] ++
-      (if isInternal then [ "witness-private" ] else [ "witness-public" ])
+    (if isInternal
+      then [ "witness-private" ]
+      else [ "witness-public" ])
   );
 
   ## We do not allocate an elastic IP for the internal witness node, so don't try to associate it
   deployment.ec2.elasticIPv4 = lib.mkIf (n == 0) (lib.mkForce "");
 
   networking.firewall.allowedTCPPorts = [
-    4040 4041   # Witness ZMQ API
+    4010 4011   # Witness ZMQ API
     4030        # Witness HTTP Wallet API
   ];
 
-  services.disciplina = {
+  services.disciplina = let
+    config-key = "alpha";
+
+  in rec {
     enable = true;
     type = "witness";
 
+    config."${config-key}".witness = rec {
+      appDir = "/var/lib/disciplina-${type}";
+      db.path = "${appDir}/witness.db";
+      api.addr = "0.0.0.0:4030";
+    };
+
     args = let
       cat = path: ''"$(cat "${path}")"'';
-      stateDir = "/var/lib/disciplina-witness";
-
-      publicIP = ''"$(curl "http://169.254.169.254/latest/meta-data/public-ipv4")"'';
-      privateIP = ''"$(curl "http://169.254.169.254/latest/meta-data/local-ipv4")"'';
+      publicIP = ''"$(curl "http://169.254.169.254/latest/meta-data/public-ipv4" 2>/dev/null)"'';
```

Comment:
what could go wrong?

Reply:
There're reasons I `set -euxo pipefail` explicitly in the service script. This is one of those reasons.